applicationset/controllers/applicationset_controller_test.go

```diff
@@ -1926,3 +1928,2090 @@ func TestSetApplicationSetStatusCondition(t *testing.T) {
+				"app1": "11",
+				"app2": "22",
+			},
+			appMap: map[string]argov1alpha1.Application{
+				"app1": {
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "app1",
+					},
+					Status: argov1alpha1.ApplicationStatus{
+						Sync: argov1alpha1.SyncStatus{
+							Status: argov1alpha1.SyncStatusCodeOutOfSync,
+						},
+					},
+				},
+				"app2": {
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "app2",
+					},
+					Status: argov1alpha1.ApplicationStatus{
+						Sync: argov1alpha1.SyncStatus{
+							Status: argov1alpha1.SyncStatusCodeOutOfSync,
+						},
+					},
+				},
+			},
+			expectedMap: map[string]bool{
+				"app1": true,
+				"app2": false,
+			},
+		},
+		{
+			name: "handles RollingSync applications that are up to date, but not healthy",
+			appSet: argov1alpha1.ApplicationSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "name",
+					Namespace: "argocd",
+				},
+				Spec: argov1alpha1.ApplicationSetSpec{
+					Strategy: &argov1alpha1.ApplicationSetStrategy{
+						Type: "RollingSync",
+					},
+				},
+				Status: argov1alpha1.ApplicationSetStatus{
+					ApplicationStatus: []argov1alpha1.ApplicationSetApplicationStatus{
+						{
+							Application:  "app1",
+							ObservedHash: "11",
+							Status:       "Progressing",
+						},
+						{
+							Application:  "app2",
+							ObservedHash: "22",
+							Status:       "Progressing",
+						},
+					},
+				},
+			},
+			appDependencyList: [][]string{
+				{"app1"},
+				{"app2"},
+			},
+			appHashMap: map[string]string{
+				"app1": "11",
+				"app2": "22",
+			},
+			expectedMap: map[string]bool{
+				"app1": true,
+				"app2": false,
+			},
+		},
+		{
+			name: "handles a lot of applications",
+			appSet: argov1alpha1.ApplicationSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "name",
+					Namespace: "argocd",
+				},
+				Spec: argov1alpha1.ApplicationSetSpec{
+					Strategy: &argov1alpha1.ApplicationSetStrategy{
+						Type: "RollingUpdate",
+					},
+				},
+				Status: argov1alpha1.ApplicationSetStatus{
+					ApplicationStatus: []argov1alpha1.ApplicationSetApplicationStatus{
+						{
+							Application:  "app1",
+							ObservedHash: "1",
+							Status:       "Healthy",
+						},
+						{
+							Application:  "app2",
+							ObservedHash: "2",
+							Status:       "Healthy",
+						},
+						{
+							Application:  "app4",
+							ObservedHash: "4",
+							Status:       "Healthy",
+						},
+						{
+							Application:  "app5",
+							ObservedHash: "5",
+							Status:       "Healthy",
+						},
+						{
+							Application:  "app7",
+							ObservedHash: "7",
+							Status:       "Healthy",
+						},
+					},
+				},
+			},
+			appDependencyList: [][]string{
+				{"app1", "app4", "app7"},
+				{"app2", "app5", "app8"},
+				{"app3", "app6", "app9"},
+			},
+			appHashMap: map[string]string{
+				"app1": "1",
+				"app2": "2",
+				"app4": "4",
+				"app7": "7",
+			},
+			expectedMap: map[string]bool{
+				"app1": true,
+				"app2": true,
```

Comment:
This test is confusing, I'll change the app numbers to be more intuitive. In this test, the first stage is complete because 1/4/7 are all 'Healthy', so both the first and second stage should be set to `true` to allow for syncing.